logging/logging.go

```diff
@@ -0,0 +1,89 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"os"
+
+	"github.com/Microsoft/go-winio"
+)
+
+// Config is passed to the binary logging function
+type Config struct {
+	ID        string
+	Namespace string
+	Stdout    io.Reader
+	Stderr    io.Reader
+}
+
+// LoggerFunc is a binary logging function signature
+type LoggerFunc func(context.Context, *Config, func() error) error
+
+// Run runs LoggerFunc
+func Run(fn LoggerFunc) {
```

Comment:
As we discussed in the standup, let's move this implementation to https://github.com/containerd/containerd/tree/master/runtime/v2/logging

Reply:
https://github.com/containerd/containerd/pull/4759